### PR TITLE
add github actions to boilerplate

### PR DESCRIPTION
**Pull request by StevenClontz**

### CODE REVIEW

File: .github/workflows/test-on-pr.yml [hunk 1]
```diff
@@ -8,7 +8,7 @@ on:
   workflow_dispatch:
 
 jobs:
-  format:
+  format-and-types:
```
Comment (StevenClontz) [inline .github/workflows/test-on-pr.yml:11]: because types keep failing here, not just formatting

File: pretext/cli.py [hunk 2]
```diff
@@ -324,7 +324,7 @@ def new(template: str, directory: Path, url_template: str) -> None:
     "files",
     help="Specify file to refresh.",
     multiple=True,
-    type=click.Choice(constants.PROJECT_RESOURCES, case_sensitive=False),
+    type=click.Choice([r for r in constants.PROJECT_RESOURCES], case_sensitive=False),
```
Comment (StevenClontz) [inline pretext/cli.py:327]: changed `constants.PROJECT_RESOURCES` to an object where keys are filenames and values are paths

File: pretext/constants.py [hunk 3]
```diff
@@ -143,10 +144,12 @@
...
 }
 
-PROJECT_RESOURCES = [
-    "project.ptx",
-    "codechat_config.yaml",
-    ".gitignore",
-    ".devcontainer.json",
-    "requirements.txt",
-]
+PROJECT_RESOURCES = {
```
Comment (StevenClontz) [inline pretext/constants.py:147]: see?!

File: templates/test-build.yml [hunk 4]
```diff
@@ -0,0 +1,35 @@
+# <!-- Managed automatically by PreTeXt authoring tools -->
+# (delete the above line to manage this file manually)
+
+name: Test Build
+on:
+    # Runs on pull requests, when a pull request is merged into main,
+    # and when a commit is otherwise pushed to main.
+    push:
```
Comment (oscarlevin) [inline templates/test-build.yml:8]: I would rather have this one not automatically run by default either.  I'm worried that a new user will be experimenting, get emails saying an action failed, and get worried.

**Review by oscarlevin — CHANGES_REQUESTED** [review]
I added one suggested change.  Besides that I think this looks really good.

Eventually we should write a better boilerplate for a README that says what all the extra files are that the CLI creates.  But that's for another time.

File: templates/test-build.yml (hunk 4 above)
Comment (StevenClontz) [reply templates/test-build.yml:8]: Can we try it like this? If you're right, we can push an update that automatically replaces this with your suggested behavior.

Comment (StevenClontz) [reply templates/test-build.yml:8]: And it'll help a lot with support. "Why won't my book build?" With this we just ask for the repo and check out the action and artifact.

**Review by oscarlevin — APPROVED** [review]

**Pull request merged**